controllers/helmchartinstall_controller.go

```diff
@@ -47,8 +52,23 @@ func (r *HelmChartInstallReconciler) Reconcile(req ctrl.Request) (ctrl.Result, e
 	ctx := context.Background()
 	_ = r.Log.WithValues("helmchartinstall", req.NamespacedName)
 
-	// your logic here
-	i := &helmv1alpha1.HelmChartInstall{}
+	// TODO NOTE the group, version, and kind would normally come from the
+	// stack configuration, and would be part of the configuration for the render
+	// callback
+	//
+	// We grab the claim as an unstructured so that we can have the same code handle
+	// arbitrary claim types. The types will be erased by this point, so if a stack
+	// author wants to validate the schema of a claim, they can do it by putting a
+	// schema in the CRD of the claim type so that the claim's schema is validated
+	// at the time that the object is accepted by the api server.
+	i := &unstructured.Unstructured{}
+	i.SetGroupVersionKind(
+		schema.GroupVersionKind{
+			Group:   "helm.samples.stacks.crossplane.io",
+			Version: "v1alpha1",
+			Kind:    "HelmChartInstall",
+		},
+	)
 	if err := r.Client.Get(ctx, req.NamespacedName, i); err != nil {
 		if kerrors.IsNotFound(err) {
```

Comment:
I'd say the general philosophy is to repeat yourself a little rather than take dependencies for something this trivial, so there's a strong case for not taking the dependency here. It might be worth copying the pattern with a TODO to use the crossplane-runtime version if/when the two projects are ever more tightly coupled.